### PR TITLE
feat(replay): add "what to watch" screen

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

People sometimes find it difficult to know where to start with replays. Our best advice is to think of a product area you want to focus on, and then drill down form there.

We can productize that advice by giving people a jumping-off point, suggestions for areas they can look into.

## Changes

Adds a "what to watch" screen with "templates" suggesting things they might want to watch.

https://github.com/user-attachments/assets/d1881ce0-ee84-4612-8c3d-3796576de677

TODO: 
- [x] Figure out why feature flags filtering doesn't work
- [x] Put this behind a flag so I can test it live
- [x] Add spacing above add button
- [x] Figure out why the variables don't hide when I tell them to hide...

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

manually 🙈

### CODE REVIEW

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts
Comment (raquelmsmith) [inline frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: lmk if you have ideas for other templates. I want to keep the list short & easy to parse, but we can decide which ones are best.

File: frontend/src/lib/lemon-ui/LemonCard/LemonCard.tsx [hunk 1]
```diff
@@ -25,6 +33,19 @@ export function LemonCard({
             onClick={onClick}
             {...props}
         >
+            {closeable ? (
+                <div className="absolute top-2 right-2">
+                    <LemonButton
+                        icon={<IconX />}
+                        onClick={(e) => {
+                            e.stopPropagation()
```
Comment (pauldambra) [inline frontend/src/lib/lemon-ui/LemonCard/LemonCard.tsx:41]: if this doesn't swallow the click then the on click handler fires as well

File: frontend/src/lib/lemon-ui/LemonCard/LemonCard.tsx [hunk 2]
```diff
@@ -25,6 +33,19 @@ export function LemonCard({
             onClick={onClick}
```
Comment (pauldambra) [inline frontend/src/lib/lemon-ui/LemonCard/LemonCard.tsx:33]: commenting here cos you can't comment on unchanged lines
we should have a different cursor when there's an onclick handler so folk can see it is clickable

File: frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+const getPageviewFilterValue = (pageview: string): UniversalFiltersGroupValue => {
+    return {
+        key: 'visited_page',
+        value: pageview,
+        operator: PropertyOperator.IContains,
+        type: PropertyFilterType.Recording,
+    }
+}
+
+export interface ReplayTemplateLogicPropsType {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx:27]: changed this from habit while figuring out what was happening

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        ],
+        categories: ['More'],
+    },
+    {
+        key: 'scattershot',
+        name: 'Scattershot',
+        description: 'Watch all recent replays, and see where users are getting stuck.',
+        variables: [],
+        categories: ['More'],
+    },
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/templates/availableTemplates.ts:132]: could replace this one with "by activity score" 

if visual regression ever lets me merge https://github.com/PostHog/posthog/pull/25701

i guess this is two points... 
one let's have activity score
two do we need this scattershot one? since it's the equivalent of "not choosing a template" 

no strong feeling but if you're looking to keep the list short

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import { ReplayTemplateType } from '~/types'
+
+export const replayTemplates: ReplayTemplateType[] = [
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/templates/availableTemplates.ts:3]: could have a "sessions with errors"

any `$exception` event or `console_error_count` > 0

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: we could add images copied from dashboard templates to the cards to make it look more like the dashboard templates list

(although arguably that's very visually busy and could be hard to parse)

Screenshot 2024-10-21 at 22 11 29

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts (hunk 4 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:132]: if we keep scattershot, it should actively set the ordering to recent
if someone has already changed sorting away from recent then they click on scattershot it doesn't reset

quite an edge case since not many folk change sorting

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: we've got "purchase flow" and "pricing page" but I think they're really the same?

could possibly lose the B2B/B2C split too but I guess then it's just one big lump

**Review by pauldambra — APPROVED** [review]
some typescript faff to solve but otherwise 🚢 

(lots of comments only reflects being excited about the feature :))

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    },
+    {
+        key: 'scattershot',
+        name: 'Scattershot',
+        description: 'Watch all recent replays, and see where users are getting stuck.',
+        variables: [],
+        categories: ['More'],
+    },
+    {
+        key: 'person-property',
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/templates/availableTemplates.ts:134]: the most common person property applied is `email` so we could prompt here to help folk understand

File: frontend/src/scenes/session-recordings/templates/availableTemplates.ts
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: could add a rage click one?

Comment (raquelmsmith) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: Oh rageclick is a good idea, will add! What about deadclicks - didn't you have some event sending for that?

Comment (raquelmsmith) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: > we could add images copied from dashboard templates

I very much dislike that UI 😅 🙈 but I will experiment with some icons.

> we've got "purchase flow" and "pricing page" but I think they're really the same?

I think they're different for b2b and b2c/ecommerce - one is when someone is viewing eg saas pricing before signup, the other is when someone is purchasing something directly (eg a pair or earrings or a hammer). I think they should be different templates.

> could possibly lose the B2B/B2C split too but I guess then it's just one big lump

I'm gonna keep the split for now to (hopefully) make it easier for people to find the things that are relevant to them. But we can see what people use and go from there.

Comment (raquelmsmith) [reply frontend/src/scenes/session-recordings/templates/availableTemplates.ts:1]: Added some icons

image

File: frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    template: ReplayTemplateType
+    // one card can be in multiple categories,
+    // key on the category so that multiple instances of the same card are isolated
+    category: ReplayTemplateCategory
+}
+
+export const sessionReplayTemplatesLogic = kea<sessionReplayTemplatesLogicType>([
+    path(() => ['scenes', 'session-recordings', 'templates', 'sessionReplayTemplatesLogic']),
+    props({} as ReplayTemplateLogicPropsType),
+    key((props) => `${props.category}-${props.template.key}`),
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx:66]: duplicate cards would share interactions so let's key on the current category as well as the template

**Review by pauldambra — APPROVED** [review]
couple of comments

but stamp so you can keep moving in case TZ overlap gets in the way 🙈

File: frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx (hunk 7 above)
Comment (raquelmsmith) [reply frontend/src/scenes/session-recordings/templates/sessionRecordingTemplatesLogic.tsx:66]: oh yeah I meant to do that, thanks!

**Comment by raquelmsmith** [comment]
automerged without full green checks, but only because of flapping, so it's fine
